internal/containers/provider/docker.go

```diff
@@ -51,6 +52,10 @@ type DockerRequest struct {
 	// ManagedVolumes is the list of volumes that should be torn down when the
 	// provider finishes execution
 	ManagedVolumes []mount.Mount
+	// PortBindings is the list of port bindings to apply to the container
+	// when it is created. The key is the port to bind to on the host, and the
+	// value is the port to bind to on the container.
+	PortBindings map[nat.Port][]nat.PortBinding
```

Comment:
We need to add a new option here for exposed ports. The type is `nat.PortSet`: https://github.com/moby/moby/blob/c2879978022db54ef21ac93feeabce0cbd75f63d/api/types/container/config.go#L51
This option will need to be plumbed through the `container.Config`, in the `ExposedPorts` field: https://github.com/chainguard-dev/terraform-provider-imagetest/blob/765d75d50bd669e49cdea767d356ece1cfd45525/internal/containers/provider/docker.go#L158-L167

We could also use the predefined type `nat.PortMap` for the `PortBindings` variable:
```suggestion
	PortBindings nat.PortMap
```